op-service/txmgr/send_state_test.go

```diff
@@ -160,3 +172,20 @@ func TestSendStateIsNotWaitingForConfirmationAfterTxUnmined(t *testing.T) {
 	sendState.TxNotMined(testHash)
 	require.False(t, sendState.IsWaitingForConfirmation())
 }
+
+// TestSendStateTimeoutAbort ensure that this will abort if it passes the tx pool timeout
+// when no successful transactions have been recorded
+func TestSendStateTimeoutAbort(t *testing.T) {
+	sendState := newSendStateWithTimeout(10 * time.Millisecond)
+	time.Sleep(20 * time.Millisecond)
```

Comment:
I mean we could inject the timeout time & force the end user to do the calculation.
I'm not really worried about these tests flaking, though it is true that the sleep isn't great.

Reply:
Passing in the deadline doesn't really help though because we still need to access the current time to check if we've passed that deadline. It's pretty standard to introduce some abstraction for the system clock to make it testable - otherwise these little sleeps in tests just keep growing over time and slowing everything down.
If it was a big change I could understand but it's such an easy win to pass in `time.Now` and use it as an abstract function within `SendState` that I'm really really keen to do it right and avoid the sleep entirely.

Reply:
Ok, I faked the clock. It's not quite as bad as I though, though I still don't love having to inject it.